pkgs/build-support/trivial-builders/test/invoke-writeMultipleReferencesToFile.nix

```diff
@@ -0,0 +1,6 @@
+{ lib ? import ../../../../lib
+, pkgs ? import ../../../.. { config = { }; overlays = { }; }
+}:
```

Comment:
I don't suppose we'll need these in the future? Can just do `callPackage` and refer to the test through the nixpkgs tree

Reply:
H'm, I currently get this:

```
❯ nix run github:ShamrockLee/nixpkgs/write-multiple-references#tests.trivial-builders.references
...

       error: getting status of '/nix/store/n1pz69ll149p06wf51k728rrwhz8aywr-source/pkgs/build-support/trivial-builders/test/samples': No such file or directory
```

Reply:
> H'm, I currently get this:
> 
> ```
> ❯ nix build github:ShamrockLee/nixpkgs/write-multiple-references#tests.trivial-builders.references
> ...
> 
>        error: getting status of '/nix/store/n1pz69ll149p06wf51k728rrwhz8aywr-source/pkgs/build-support/trivial-builders/test/samples': No such file or directory
> ```

Oops! It should be `samples.nix` instead of `samples`.

Reply:
AFAIU I'm getting an unexpected extra line:

```
vm-test-run-nixpkgs-trivial-builders> machine: output: --- /dev/fd/63   2023-11-29 13:38:10.831614073 +0000
vm-test-run-nixpkgs-trivial-builders> +++ /dev/fd/62    2023-11-29 13:38:10.831614073 +0000
vm-test-run-nixpkgs-trivial-builders> @@ -14,3 +14,4 @@
vm-test-run-nixpkgs-trivial-builders>  /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27
vm-test-run-nixpkgs-trivial-builders>  /nix/store/s2f1sqfsdi4pmh23nfnrh42v17zsvi5y-libunistring-1.1
vm-test-run-nixpkgs-trivial-builders>  /nix/store/sbldylj3clbkc0aqvjjzfa6slp4zdvlj-hello-2.12.1
vm-test-run-nixpkgs-trivial-builders> +/nix/store/wxiv7cbgpncmc87qadf14nnf9ip39749-invoke-writeReferencesToFile.nix
```

Reply:
> AFAIU I'm getting an unexpected extra line:
> 
> ```
> vm-test-run-nixpkgs-trivial-builders> machine: output: --- /dev/fd/63   2023-11-29 13:38:10.831614073 +0000
> vm-test-run-nixpkgs-trivial-builders> +++ /dev/fd/62    2023-11-29 13:38:10.831614073 +0000
> vm-test-run-nixpkgs-trivial-builders> @@ -14,3 +14,4 @@
> vm-test-run-nixpkgs-trivial-builders>  /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27
> vm-test-run-nixpkgs-trivial-builders>  /nix/store/s2f1sqfsdi4pmh23nfnrh42v17zsvi5y-libunistring-1.1
> vm-test-run-nixpkgs-trivial-builders>  /nix/store/sbldylj3clbkc0aqvjjzfa6slp4zdvlj-hello-2.12.1
> vm-test-run-nixpkgs-trivial-builders> +/nix/store/wxiv7cbgpncmc87qadf14nnf9ip39749-invoke-writeReferencesToFile.nix
> ```

This means that the line was missing in the file produced by `writeMultipleReferencesToFile`.

After hours of debugging, it turns out that `toString` doesn't always do string interpolation on string-interoperable stuff, especially when the input is a [path value](https://nixos.org/manual/nix/stable/language/values#type-path).

Change the `toString` to `(path: "${path}")` that constructs the `pathListFile`, and the problem is solved.

Reply:
> I don't suppose we'll need these in the future? Can just do `callPackage` and refer to the test through the nixpkgs tree

They are there to make it possible to `nix-build` the expression file directly, in the hope that we could run `references-test.sh` directly without having to spawn a VM to run the tests.

This currently fails, as stated in the `references-test.sh`. I think that the failure is caused by bugs yet to be discovered instead of intrinsic limitation, and I keep it in the hope that it would work in the future.

Reply:
Uh-huh. So

1. The motivation for calling the file directly as opposed to through `test.trivial-builders.references` is to be able to run the `.sh` script outside the VM
2. Tests outside the VM do not work yet

RE: 1. Can't we just passthru the `invoke*` derivation in the `nixosTest`? Then we could access it by the attribute path

RE: 2. Let's just merge the part that is usable right now, and carry over the less clean bits into the later PRs?

Reply:
> RE: 1. Can't we just passthru the `invoke*` derivation in the `nixosTest`? Then we could access it by the attribute path

That's sounds reasonable. I implemented it in #204462, which becomes a great refactor of `tests.trivial-builders.references`. I'll try to add the tests here with minimal changes, and integrate the two PRs once any of them get merged.